streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -235,87 +226,99 @@
                     Type.LIST,
                     Importance.HIGH,
                     CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
-            .define(CLIENT_ID_CONFIG,
```

Comment:
Thanks for the explanation: checking per-commit is indeed easier.

Moving forward we can even make them two separate PRs for other reviewers to easily review.